official/recommendation/data_download.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Download and extract the movielens dataset from grouplens website.
+
+Download the dataset, and perform data-preprocessing to convert the raw dataset
+into csv file to be used in model training and evaluation.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import collections
+import os
+import sys
+import time
+import urllib
+import zipfile
+
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+import numpy as np
+import pandas as pd
```

Comment:
Not sure if it worth the effort or not, maybe we can convert this to use the tf.data for importing CSV files.